### PR TITLE
nfs: configure RGW FSAL to start up correctly

### PR DESCRIPTION
**Pull request by alimaredia**
- Add RGW keyring to nfs node
- Add RGW section to ganesha.conf
- Add RGW section to ceph.conf onf nfs node

Signed-off-by: Ali Maredia <amaredia@redhat.com>

### CODE REVIEW

**Comment by alimaredia** [comment]
@leseb The RGW FSAL wasn't being properly started in nfs-ganesha before this PR. The developers that created the RGW FSAL for nfs-ganesha have always had ganesha running on the same machines as the RGW, so much of this PR is doing that.

The ganesha config file needs a ceph.conf to read from, and the ceph.conf on the nfs node did not have a section similar to the rgw section on the rgw nodes as well as rgw admin keys. This PR only enables all of that extra config and admin key functionality

Below is the ceph.conf with valid entries that gets generated on an nfs node with nfs_obj_gw set to true: 

https://paste.fedoraproject.org/paste/44GgmshCXkxr~zWoiS2VEg

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-bluestore_docker_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-purge_dmcrypt

File: roles/ceph-nfs/templates/ganesha.conf.j2 [hunk 1]
```diff
@@ -52,6 +52,13 @@ EXPORT
         {{ ganesha_rgw_export_overrides | default(None) }}
 
 }
+
+RGW {
+        ceph_conf = "/etc/ceph/{{ cluster }}.conf";
+        cluster = "{{ cluster }}";
+        name = "{{ rgw_client_name }}";
```
Comment (alimaredia) [inline roles/ceph-nfs/templates/ganesha.conf.j2]: @leseb name here is the same as "client.rgw.nfs0" here: https://paste.fedoraproject.org/paste/44GgmshCXkxr~zWoiS2VEg

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Comment by GregMeno** [comment]
jenkins test luminous-ansible2.3-docker_cluster

**Comment by alimaredia** [comment]
@andrewschoen `generate_group_vars_sample.sh` was run and I added a new test to verify the functionality. We should be good to go.

**Comment by alimaredia** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Comment by alimaredia** [comment]
jenkins test luminous-ansible2.3-docker_cluster

**Comment by alimaredia** [comment]
jenkins test luminous-ansible2.3-purge_dmcrypt

**Comment by andrewschoen** [comment]
@alimaredia I'm seeing this failure in the tests

```
=================================== FAILURES ===================================
____________ TestNFSs.test_nfs_rgw_fsal_export[ansible://ceph-nfs0] ____________
[gw1] linux2 -- Python 2.7.13 /tmp/tox.mNIMDPHJyT/luminous-ansible2.3-update_cluster/bin/python2

self = <test_nfs_ganesha.TestNFSs object at 0x7f8dab140e50>
node = {'address': '192.168.1.60', 'ceph_stable_release': 'luminous', 'cluster_address': '', 'cluster_name': u'test', ...}
host = <testinfra.host.Host object at 0x7f8da80eb450>

    @pytest.mark.no_docker
    def test_nfs_rgw_fsal_export(self, node, host):
        cmd = host.run("sudo mount.nfs localhost:/ceph /mnt/")
>       assert cmd.rc == 0
E       assert 32 == 0
E        +  where 32 = CommandResult(command=u"sudo /bin/sh -c 'sudo mount.nfs localhost:/ceph /mnt/'...xit_status=32, stdout=u'', stderr='mount.nfs: /mnt is busy or already mounted').rc
```

This failure is from an ``update_cluster`` scenario: https://2.jenkins.ceph.com/job/ceph-ansible-prs-luminous-ansible2.3-update_cluster/418/console

I think the issue is that when the testinfra tests run again after the rolling update the mount already exists and the test fails. The best approach here is to have a ``teardown`` method that removes the mount after the test has ran.

**Comment by alimaredia** [comment]
@andrewschoen as long as the test i added looks good, this should be ready to merge.

**Comment by andrewschoen** [comment]
jenkins test luminous-ansible2.3-purge_cluster

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-switch_to_containers

**Comment by leseb** [comment]
jenkins test luminous-ansible2.3-purge_cluster

**Pull request merged**